main/src/ca/uwaterloo/flix/language/ast/Type.scala

```diff
@@ -301,25 +274,60 @@ object Type {
       * The kind of a type application can unique be determined
       * from the kind of the first type argument `t1`.
       */
-    def kind: Kind = tpe1.kind match {
-      case Kind.Arrow(_, k) => k
-      case _ => throw InternalCompilerException("Illegal kind.")
+    def kind: Kind = {
```

Comment:
I guess there is a very interesting question (similar to the pretty print functions) about when certain functions can be called. I am not sure what the solution, but in the current "way of things" you have to think hard about when you can call `.kind`.